subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,54 @@
+//
+// Subworkflow with functionality specific to the nf-core/mag pipeline
+//
+
+workflow SAMPLESHEET_MAG {
+    take:
+    ch_processed_reads
+
+    main:
+    format     = 'csv' // most common format in nf-core
+    format_sep = ','
+
+
+    ch_list_for_samplesheet = ch_processed_reads
+            .filter { meta, sample_id, instrument_platform,fastq_1,fastq_2,fasta -> (fastq_1 && fastq_2) && !fasta }.view()
+                .map {
+                        meta, sample_id, instrument_platform,fastq_1,fastq_2,fasta ->
+                            def sample        = meta.id
+                            def run           = meta.run_accession  //this should be optional
+                            def group         = ""
+                            def short_reads_1 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_1.getName()
+                            def short_reads_2 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_2.getName()
```

Comment:
Can you not also make this support single_end using the `meta.single_end`  information?

Or have we lost that information at this stage?

```suggestion
                            def short_reads_1 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_1.getName()
                            def short_reads_2 = !meta.single_end ? file(params.outdir).toString() + '/' + meta.id + '/' + fastq_2.getName() : ""
```

Reply:
I do not follow here. We said to filter out the single ends and fasta. Or do I misunderstand something?

Reply:
Ah yeah, good point I forgot and there wasn't a comment why :sweat_smile: 

Please leave a comment next to the line saying only PE supported now, and I'll look into fixing that later :)

Reply:
@sofstam did you see what @Joon-Klaps did in detaxizer for SE/PE reads? You can copy the same concept here too  (basically to branch SE/PE separately, and call the `channelToSamplesheet` function twice ;)